x-pack/plugins/aiops/public/components/spike_analysis_table/spike_analysis_table_groups.tsx

```diff
@@ -252,8 +253,11 @@ export const SpikeAnalysisGroupsTable: FC<SpikeAnalysisTableProps> = ({
       ),
       render: (_, { group, repeatedValues }) => {
         const valuesBadges = [];
+        const hasExtraBadges = Object.keys(group).length > MAX_GROUP_BADGES;
+
         for (const fieldName in group) {
           if (group.hasOwnProperty(fieldName)) {
+            if (valuesBadges.length === MAX_GROUP_BADGES) break;
```

Comment:
Nit: If you move that line with the `break` below the `.push()` it should break after the last necessary push instead of triggering one more loop